tests/hooks/utils/conversion-utils.spec.ts

```diff
@@ -1,35 +1,93 @@
 import fs from "fs";
 import path from "path";
-import { describe, expect, test } from "vitest";
+import { beforeEach, describe, expect, test, vi } from "vitest";
 import { writeJSONToFile } from "../../../hooks/utils/conversion-utils";
 
 describe("writeJSONToFile", () => {
-  test("write a dummy json to a file", async () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.mock("fs");
+  });
+
+  test("write a file in an unexistent folder", async () => {
+    fs.promises.stat = vi.fn().mockImplementation(() => {
+      throw new Error("Mocked error");
+    });
+    fs.promises.mkdir = vi.fn().mockImplementation(() => true);
+    fs.promises.writeFile = vi.fn().mockImplementation(() => true);
+
     const mock = [
       {
-        name: "Main",
-        description: "The main Qiskit packages maintained by IBM Quantum",
-      },
-      {
-        name: "Extensions",
-        description: "IBM Quantum supported Qiskit extensions",
+        name: "Abby Mitchell",
```

Comment:
It's a good idea to avoid real names in mock data.

Reply:
Done